docs/content/docs/try-flink/write_flink_program_with_sql.md

```diff
@@ -0,0 +1,269 @@
+       COUNT(game_name) 
+FROM twitch_stream
+WHERE location IS NOT NULL
+GROUP BY location, game_name;
+```
+
+You should see a table result of continously changing data that tells you the most popular games for each country and the count for each game. 
+
+For the last example, let's try a very commonly used function in streaming SQL: computing windowed aggregations.
+
+Windows are the fundamental building blocks when it comes to processing infinite streams since they provide a way to split streams into finite chunks. 
+Flink SQL offers several [window functions]({{< ref "docs/dev/table/sql/queries/window-tvf" >}}) that you can use.  
+
+Let's explore how to apply one of these functions! To calculate an aggregation per game of the total viewer count for every 1 minute window, 
+you can use the [tumble]({{< ref "docs/dev/table/sql/queries/window-tvf#tumble" >}}) function.
+
+Tumbling windows can be thought of as mini-batches of aggregations over a non-overlapping window of time.
+
+```sql
+SELECT window_start,
+       window_end,
+       game_name,
+       SUM(viewer_count) AS TotalViewerCount
+  FROM TABLE(
+    TUMBLE(TABLE twitch_stream, DESCRIPTOR(proctime), INTERVAL '1' MINUTE))
+  GROUP BY window_start, window_end, game_name;
+```
+
+Fresh results will appear every minute, showing the sum of the viewer counts for each game.
+
+
+## Step 7: Write the updated stream to persistant storage
+
+Now that you have written a program with Flink SQL to process all that streaming data, you probably want to store it somewhere. Since you are dealing with unbounded data sets, you can think of it as maintaining materialized views (or snapshots of the data) in external storage systems.
+
+Flink does not provide its own storage system but instead offers many sink connectors you can use to write table updates to external systems.
+
+To write a table to a CSV file, you can use the [FileSystem connector]({{< ref "docs/connectors/table/filesystem" >}}) that is built into Flink. This connector supports row-encoded and bulk-encoded formats. Row-encoded formats such as CSV and JSON write one row at a time to a file. Bulk-encoded formats collect a batch of rows in memory and organize them in a storage-and-scan-efficient format before writing out the data. 
+
+To store the results of the last example query (tumble window), you need to create a table that will store the results:
+
+
+```sql
+CREATE TABLE twitch_stream_viewer_count (
+  `window_start` TIMESTAMP_LTZ(3),
+  `window_end` TIMESTAMP_LTZ(3),
+  `game_name` VARCHAR(2147483647),
+  `TotalViewerCount` INT
+) WITH (
+  'connector' = 'filesystem',
+  'path' = 'file:///path/to/directory/results.csv',
+  'format' = 'csv'
+)
+```
+
+Now insert the streaming SQL results into the file system table:
+
+```sql
+INSERT INTO twitch_stream_viewer_count 
+SELECT 
+  `window_start` TIMESTAMP_LTZ(3),
+  `window_end` TIMESTAMP_LTZ(3),
+  `game_name` VARCHAR(2147483647),
+  `TotalViewerCount` INT
+FROM twitch_stream;
```

Comment:
This insert statement should be

```suggestion
INSERT INTO twitch_stream_viewer_count 
SELECT 
  `window_start` TIMESTAMP_LTZ(3),
  `window_end` TIMESTAMP_LTZ(3),
  `game_name` VARCHAR(2147483647),
  `TotalViewerCount` INT
FROM TABLE(
    TUMBLE(TABLE twitch_stream, DESCRIPTOR(proctime), INTERVAL '1' MINUTE))
GROUP BY window_start, window_end, game_name;
```

However, this INSERT INTO is problematic, because the filesystem sink actually creates a directory full of rolling files, and those files are only finalized during checkpointing, which isn't happening here. 

In my case, I ended up with a file named `/tmp/results.csv/.part-7662f404-d437-4351-8f64-7f71944fb9f4-0-0.inprogress.0354b5f8-0c13-4c40-9fcd-d5db0ca2bb44` containing the in-progress results. 

This part of the tutorial needs more thought -- I don't think we want to do anything this confusing.

Reply:
@AHeise Is there anything in the near future that improve this? We need an option that reads and write single files for testing, demos etc. This is an early blocker that got reported multiple times.

Reply:
Perhaps leave it this way and then revisit it once we have that feature available?